test/suite/test_search_near02.py

```diff
@@ -0,0 +1,144 @@
+        if self.key_format == 'u':
+            return key.encode()
+        elif self.key_format == '10s':
+            return key.ljust(10, "\x00")
+        else:
+            return key
+
+    scenarios = make_scenarios(key_format_values)
+
+    def test_search_near(self):
+        uri = 'table:test_base_scenario'
+        self.session.create(uri, 'key_format={},value_format=S'.format(self.key_format))
+        cursor = self.session.open_cursor(uri)
+        cursor2 = self.session.open_cursor(uri, None, "debug=(release_evict=true)")
+
+        # Basic character array.
+        l = "abcdefghijklmnopqrstuvwxyz"
+
+        # Insert keys aaa -> aaz with timestamp 200.
+        prefix = "aa"
+        self.session.begin_transaction()
+        for k in range (0, 25):
+            cursor[prefix + l[k]] = prefix + l[k]
+        self.session.commit_transaction('commit_timestamp=' + self.timestamp_str(200))
+
+        # Insert key aaz timestamp 50.
+        self.session.begin_transaction()
+        cursor[prefix + "z"] = prefix + "z"
+        self.session.commit_transaction('commit_timestamp=' + self.timestamp_str(50))
+
+        # Evict the whole range.
+        for k in range (0, 26):
+            cursor2.set_key(prefix + l[k])
+            self.assertEqual(cursor2.search(), 0)
+            self.assertEqual(cursor2.reset(), 0)
+
+        # Start a transaction at timestamp 100, aaz should be the only key that is visible.
+        self.session.begin_transaction('read_timestamp=' + self.timestamp_str(100))
+        cursor3 = self.session.open_cursor(uri)
+        cursor3.set_key("aa")
+        self.assertEqual(cursor3.search_near(), 1)
+        self.assertEqual(cursor3.get_key(), self.check_key("aaz"))
+
+        cursor3.set_key("az")
+        self.assertEqual(cursor3.search_near(), -1)
+        self.assertEqual(cursor3.get_key(), self.check_key("aaz"))
+
+        cursor3.reconfigure("prefix_key=true")
+        cursor3.set_key("aa")
+        self.assertEqual(cursor3.search_near(), 1)
+        self.assertEqual(cursor3.get_key(), self.check_key("aaz"))
+
+        cursor3.set_key("aaz")
+        self.assertEqual(cursor3.search_near(), 0)
+        self.assertEqual(cursor3.get_key(), self.check_key("aaz"))
+
+        # FIXME-WT-8044 Prefix search near returns a valid key when expecting WT_NOTFOUND.
+        # cursor3.set_key("az")
+        # self.assertEqual(cursor3.search_near(), wiredtiger.WT_NOTFOUND)
+        cursor3.close()
+        self.session.commit_transaction()
+
+        # Start a transaction at timestamp 25, no keys are visible.
+        self.session.begin_transaction('read_timestamp=' + self.timestamp_str(25))
+        cursor3 = self.session.open_cursor(uri)
+        cursor3.set_key("aa")
+        self.assertEqual(cursor3.search_near(), wiredtiger.WT_NOTFOUND)
+
+        cursor3.reconfigure("prefix_key=true")
+        cursor3.set_key("aa")
+        self.assertEqual(cursor3.search_near(), wiredtiger.WT_NOTFOUND)
+
+        cursor3.set_key("aaz")
```

Comment:
Instead of tying one key, we could just try them all, I was thinking something that looks like:
```
        for k in range (0, 26):
            key = prefix + l[k]
            cursor3.set_key(key)
            self.assertEqual(cursor3.search_near(), wiredtiger.WT_NOTFOUND)
            
            cursor3.reconfigure("prefix_key=true")
            cursor3.set_key(key)
            self.assertEqual(cursor3.search_near(), wiredtiger.WT_NOTFOUND)
```